### ISSUE TITLE
HTTP/1.0 response body truncated from Cohttp_lwt_unix.Client.get

### ISSUE THREAD

**Issue by christianpanton** [mirage/ocaml-cohttp#229]
I encountered a problem using Cohttp_lwt_unix.Client.get, where the body becomes truncated if the server serves a too large document. 

I am just getting started with OCaml, so I am sorry if this POC looks funky.

```ocaml
open Lwt

let body_length = "http://u.hyg.li:5000" (* sends HTTP/1.0 response with 10.000 A's*)
                |> Uri.of_string 
                |> Cohttp_lwt_unix.Client.get 
                >>= (fun (_, body) -> return body)
                >>= Cohttp_lwt_body.to_string
                |> Lwt_main.run
                |> String.length in

Printf.printf "Response length: %d, expected: 10000\n" body_length
```

This returns `Response length: 4051, expected: 10000`. If the size of the header is included the total response size is 4096. Is that some fixed sized buffer?

The URL is temporary and might disappear when this issue is closed.

**Comment by marklrh** [comment]
Thank you for reporting this issue.

Interesting. The length of the body I got is ``1183`` instead of your ``4051``. Besides, I output the response header in s-expression format:
````
((encoding Unknown)(headers((content-type text/plain)))(version HTTP_1_0)(status OK)(flush false))
````
Seems like HTTP response header does not include ``content-length``.

**Comment by christianpanton** [comment]
Thank you for confirming this. I suspect the lack of `Content-length` to be the problem as well. Sadly, including such a header is out of my control.

However, if I read RFC2616 (defining HTTP/1.1), section 4.4 correctly, the `Message Length` can be defined in five different ways, where the last resort is simply closing the connection. So I think that this applies to HTTP/1.1 as well.

**Comment by marklrh** [comment]
1. 

> So I think that this applies to HTTP/1.1 as well.

Do you mean "HTTP/1.0"? 

2.

there is one [article](http://www8.org/w8-papers/5c-protocols/key/key.html#Message_transmission) addressed the problem

>
Message transmission

>
HTTP messages may carry a body of arbitrary length. The recipient of a message needs to know where the message ends. The sender can use the Content-Length header, which gives the length of the body. However, many responses are generated dynamically, by CGI [CGI98] processes and similar mechanisms. Without buffering the entire response (which would add latency), the server cannot know how long it will be and cannot send a Content-Length header.
>
When not using persistent connections, the solution is simple: the server closes the connection. This option is available in HTTP/1.1, but it defeats the performance advantages of persistent connections.

3.

Notice that ``curl`` can receive 10000 bytes of 'A'. 

In conclusion, there might be something wrong with Cohttp's implementation. Let's see how core developers respond with this issue.

**Comment by avsm** [comment]
We do need to keep reading until the other side sends us an EOF.  Unfortunately that can result in a permanently hung connection.

**Comment by rgrinberg** [comment]
@avsm I don't think that should matter to us. We can always add a timeout option or the user can easily implement one.

**Comment by rgrinberg** [comment]
@christianpanton is still a problem for you?

**Comment by christianpanton** [comment]
it works \o/ - thank you
